### ISSUE TITLE
SIGBUS ERROR

### ISSUE THREAD

**Issue by aak2075** [corretto/corretto-17#157]
While using IntelliJ's profiling feature, I encountered the following error:

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGBUS (0xa) at pc=0x0000000111da94e4, pid=2409, tid=8707
#
# JRE version: OpenJDK Runtime Environment Corretto-17.0.7.7.1 (17.0.7+7) (build 17.0.7+7-LTS)
# Java VM: OpenJDK 64-Bit Server VM Corretto-17.0.7.7.1 (17.0.7+7-LTS, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, bsd-aarch64)
# Problematic frame:
# v  ~StubRoutines::SafeFetch32
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# JFR recording file will be written. Location: /Users/igyuseong/IdeaProjects/2023-haru-study/backend/hs_err_pid2409.jfr
#
# If you would like to submit a bug report, please visit:
#   https://github.com/corretto/corretto-17/issues/
#

---------------  S U M M A R Y ------------

Command Line: -Dorg.gradle.internal.worker.tmpdir=/Users/igyuseong/IdeaProjects/2023-haru-study/backend/build/tmp/test/work -Dorg.gradle.native=false -agentpath:/private/var/folders/xt/49hx3tgs24q621rv0160ztm80000gp/T/libasyncProfiler.dylib=start,jfr,event=wall,interval=10ms,jfrsync=profile,cstack=no,file=/Users/igyuseong/IdeaSnapshots/Tests_in__backend_test__2023_11_15_092947.jfr,log=/private/var/folders/xt/49hx3tgs24q621rv0160ztm80000gp/T/Tests_in__backend_test__2023_11_15_092947.jfr.log.txt,logLevel=DEBUG -Xmx512m -Dfile.encoding=UTF-8 -Duser.country=KR -Duser.language=ko -Duser.variant -ea worker.org.gradle.process.internal.worker.GradleWorkerMain 'Gradle Test Executor 82'

Host: "MacBookAir10,1" arm64, 8 cores, 16G, Darwin 22.5.0, macOS 13.4 (22F66)
Time: Wed Nov 15 09:29:48 2023 KST elapsed time: 0.326611 seconds (0d 0h 0m 0s)

---------------  T H R E A D  ---------------

Current thread (0x000000015508d800):  JavaThread "main" [_thread_in_Java, id=8707, stack(0x000000016f1f4000,0x000000016f3f7000)]

Stack: [0x000000016f1f4000,0x000000016f3f7000],  sp=0x000000016f3f5b10,  free space=2054k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
v  ~StubRoutines::SafeFetch32
V  [libjvm.dylib+0x7c0254]  os::is_readable_range(void const*, void const*)+0x2c
V  [libjvm.dylib+0x76d01c]  Method::is_valid_method(Method const*)+0x38
V  [libjvm.dylib+0x3620ec]  frame::is_interpreted_frame_valid(JavaThread*) const+0x7c
V  [libjvm.dylib+0x361fc4]  frame::safe_for_sender(JavaThread*)+0x504
V  [libjvm.dylib+0x9b9a8c]  JavaThread::pd_get_top_frame(frame*, void*, bool)+0x1f4
V  [libjvm.dylib+0x35ecd0]  AsyncGetCallTrace+0x1a4
C  [libasyncProfiler.dylib+0x3e400]  Profiler::getJavaTraceAsync(void*, ASGCT_CallFrame*, int, StackContext*)+0x15c
C  [libasyncProfiler.dylib+0x3f518]  Profiler::recordSampleForThread(int, void*, unsigned long long, EventType, Event*, int*)+0x15c
C  [libasyncProfiler.dylib+0x4e114]  WallClock::signalHandler(int, __siginfo*, void*)+0xa8
C  [libsystem_platform.dylib+0x3a24]  _sigtramp+0x38
C  0xb949800111db17d8
j  java.lang.Thread.setPriority(I)V+52 java.base@17.0.7
j  java.lang.Thread.<init>(Ljava/lang/ThreadGroup;Ljava/lang/Runnable;Ljava/lang/String;JLjava/security/AccessControlContext;Z)V+207 java.base@17.0.7
j  java.lang.Thread.<init>(Ljava/lang/ThreadGroup;Ljava/lang/Runnable;Ljava/lang/String;J)V+8 java.base@17.0.7
j  java.lang.Thread.<init>(Ljava/lang/String;)V+5 java.base@17.0.7
v  ~StubRoutines::call_stub
V  [libjvm.dylib+0x47225c]  JavaCalls::call_helper(JavaValue*, methodHandle const&, JavaCallArguments*, JavaThread*)+0x390
V  [libjvm.dylib+0x4cdf34]  jni_invoke_nonstatic(JNIEnv_*, JavaValue*, _jobject*, JNICallType, _jmethodID*, JNI_ArgumentPusher*, JavaThread*)+0x2cc
V  [libjvm.dylib+0x4ce174]  jni_NewObjectV+0x194
C  [libasyncProfiler.dylib+0x44218]  JNIEnv_::NewObject(_jclass*, _jmethodID*, ...)+0x24
C  [libasyncProfiler.dylib+0x41f00]  Profiler::startTimer()+0xf4
C  [libasyncProfiler.dylib+0x41810]  Profiler::start(Arguments&, bool)+0xc2c
C  [libasyncProfiler.dylib+0x443f4]  Profiler::runInternal(Arguments&, std::__1::basic_ostream<char, std::__1::char_traits<char>>&)+0x6c
C  [libasyncProfiler.dylib+0x4be40]  VM::VMInit(_jvmtiEnv*, JNIEnv_*, _jobject*)+0xe0
V  [libjvm.dylib+0x634270]  JvmtiExport::post_vm_initialized()+0x240
V  [libjvm.dylib+0x9ac928]  Threads::create_vm(JavaVMInitArgs*, bool*)+0x874
V  [libjvm.dylib+0x4e99ec]  JNI_CreateJavaVM+0x78
C  [libjli.dylib+0x6e70]  JavaMain+0x108
C  [libjli.dylib+0x9ab8]  ThreadJavaMain+0xc
C  [libsystem_pthread.dylib+0x6fa8]  _pthread_start+0x94


siginfo: si_signo: 10 (SIGBUS), si_code: 1 (BUS_ADRALN), si_addr: 0x0000000111da94e4

Register to memory mapping:

 x0=0x00000008003f9000 is pointing into metadata
 x1=0x00000000cafebabe is an unknown value
 x2=0x000000016f3f5c78 is pointing into the stack for thread: 0x000000015508d800
 x3=0x0000000000000001 is an unknown value
 x4=0x000000016f3f5e50 is pointing into the stack for thread: 0x000000015508d800
 x5=0x000000016f3f5e50 is pointing into the stack for thread: 0x000000015508d800
 x6=0x0000000044000002 is an unknown value
 x7=0x0000000000000001 is an unknown value
 x8=0x0000000111da94e0 is at begin+0 in a stub
StubRoutines::SafeFetch32 [0x0000000111da94e0, 0x0000000111da94ec] (12 bytes)
 x9=0x000000016f3f6400 is pointing into the stack for thread: 0x000000015508d800
x10=0x0 is NULL
x11=0x000000016f3f6420 is pointing into the stack for thread: 0x000000015508d800
x12=0x00000000000000ab is an unknown value
x13=0x0 is NULL
x14=0x00000000000007fb is an unknown value
x15=0x00000000b3c7affb is an unknown value
x16=0x0000000197742e10: pthread_getspecific+0 in /usr/lib/system/libsystem_pthread.dylib at 0x0000000197741000
x17=0x0000600002389650 points into unknown readable memory: 0xfffffffffffffffe | fe ff ff ff ff ff ff ff
x18=0x0 is NULL
x19=0x00000008003f9000 is pointing into metadata
x20=0x000000015508d800 is a thread
x21=0x0000000000000001 is an unknown value
x22=0x000000016f3f5c90 is pointing into the stack for thread: 0x000000015508d800
x23=0x0000000102a279b0: _ZN12StubRoutines18_safefetch32_entryE+0 in /Users/igyuseong/Library/Java/JavaVirtualMachines/corretto-17.0.7/Contents/Home/lib/server/libjvm.dylib at 0x0000000101dc0000
x24=0x00000000cafebabe is an unknown value
x25=0x000000016f3f6470 is pointing into the stack for thread: 0x000000015508d800
x26=0x000000016f3f6400 is pointing into the stack for thread: 0x000000015508d800
x27=0x00000001480143d8 points into unknown readable memory: 0x0000000000000126 | 26 01 00 00 00 00 00 00
x28=0x0000000111db1804 is at code_begin+516 in an Interpreter codelet
method entry point (kind = native)  [0x0000000111db1600, 0x0000000111db1c80]  1664 bytes


Registers:
 x0=0x00000008003f9000  x1=0x00000000cafebabe  x2=0x000000016f3f5c78  x3=0x0000000000000001
 x4=0x000000016f3f5e50  x5=0x000000016f3f5e50  x6=0x0000000044000002  x7=0x0000000000000001
 x8=0x0000000111da94e0  x9=0x000000016f3f6400 x10=0x0000000000000000 x11=0x000000016f3f6420
x12=0x00000000000000ab x13=0x0000000000000000 x14=0x00000000000007fb x15=0x00000000b3c7affb
x16=0x0000000197742e10 x17=0x0000600002389650 x18=0x0000000000000000 x19=0x00000008003f9000
x20=0x000000015508d800 x21=0x0000000000000001 x22=0x000000016f3f5c90 x23=0x0000000102a279b0
x24=0x00000000cafebabe x25=0x000000016f3f6470 x26=0x000000016f3f6400 x27=0x00000001480143d8
x28=0x0000000111db1804  fp=0x000000016f3f5b40  lr=0x000000010257fc54  sp=0x000000016f3f5b10
pc=0x0000000111da94e4 cpsr=0x0000000060001000
Top of Stack: (sp=0x000000016f3f5b10)
0x000000016f3f5b10:   000000016f3f63e0 0000000102a27758
0x000000016f3f5b20:   000000016f3f5c90 00000008003f9650
0x000000016f3f5b30:   00000008003f9000 00000008003f96a8
0x000000016f3f5b40:   000000016f3f5b60 0000000102580254
0x000000016f3f5b50:   000000016f3f5bc0 00000008003f9650
0x000000016f3f5b60:   000000016f3f5b80 000000010252d01c
0x000000016f3f5b70:   000000016f3f5bc0 000000015508d800
0x000000016f3f5b80:   000000016f3f5bb0 00000001021220ec
0x000000016f3f5b90:   000000016f3f5c90 0000000111dad590
0x000000016f3f5ba0:   0000000111dadfc8 000000015508d800
0x000000016f3f5bb0:   000000016f3f5c30 0000000102121fc4
0x000000016f3f5bc0:   000000016f3f6400 0000000111dadfc8
0x000000016f3f5bd0:   0000000111dad590 0000000000000000
0x000000016f3f5be0:   000000016f3f6470 000000016f3f63e0
0x000000016f3f5bf0:   0000000148010364 000000015508daa8
0x000000016f3f5c00:   000000015508d800 000000016f3f5e50
0x000000016f3f5c10:   0000000111dad590 0000000111db1804
0x000000016f3f5c20:   000000015508d800 000000016f3f5d40
0x000000016f3f5c30:   000000016f3f5cd0 0000000102779a8c
0x000000016f3f5c40:   000000015516a620 000000011d236b10
0x000000016f3f5c50:   000000011d06b928 000000015508d800
0x000000016f3f5c60:   0000000600000003 4d4a47dd00000024
0x000000016f3f5c70:   000000016f3f6390 000000016f3f63f0
0x000000016f3f5c80:   000000016f3f6390 0000000111db1804
0x000000016f3f5c90:   0000000111dad590 0000000800000000
0x000000016f3f5ca0:   000000016f3f63f0 000000016f3f6390
0x000000016f3f5cb0:   000000016f3f5fa8 0000000000000800
0x000000016f3f5cc0:   000000015508d800 000000016f3f5db0
0x000000016f3f5cd0:   000000016f3f5d90 000000010211ecd0
0x000000016f3f5ce0:   0000000000000001 000000015508d800
0x000000016f3f5cf0:   000000016f3f5e00 0000000102735a50
0x000000016f3f5d00:   000000016f3f5f40 00000000ffffffff 

Instructions: (pc=0x0000000111da94e4)
0x0000000111da93e4:   1e614000 1400003a d29a2809 f2a053e9
0x0000000111da93f4:   f2c00029 6d425935 1e640886 0c402d31
0x0000000111da9404:   1e6408c7 1f5654d0 1e6c1017 1f5050d0
0x0000000111da9414:   1f504cd0 1f5048d0 1e650af6 1f50d8f6
0x0000000111da9424:   1f5694d6 1f5158f6 1e762880 36000501
0x0000000111da9434:   1e614000 14000026 2e251ca5 d29a3009
0x0000000111da9444:   f2a053e9 f2c00029 6d425d36 1e600806
0x0000000111da9454:   0c402d32 1f5758c7 1e6c101a 1f4754c7
0x0000000111da9464:   1e650805 1f4750c7 d2866668 f2a7fa68
0x0000000111da9474:   1f474cc7 1e6608c0 1f4748c7 1e6e1019
0x0000000111da9484:   eb08007f 540000ac 1f479400 1f460340
0x0000000111da9494:   1e603b20 1400000e d2a7fd29 eb09007f
0x0000000111da94a4:   540000ac 51480069 d3607d29 9e670121
0x0000000111da94b4:   14000002 1e6a5001 1f679417 1f668742
0x0000000111da94c4:   1e613b23 1e773857 1e773860 a8c17ff3
0x0000000111da94d4:   910003bf a8c17bfd d65f03c0 b9400001
0x0000000111da94e4:   aa0103e0 d65f03c0 f9400001 aa0103e0
0x0000000111da94f4:   d65f03c0 00000000 00000000 00000000
0x0000000111da9504:   00000000 00000000 00000000 00000000
0x0000000111da9514:   00000000 00000000 00000000 00000000
0x0000000111da9524:   00000000 00000000 00000000 00000000
0x0000000111da9534:   00000000 00000000 00000000 00000000
0x0000000111da9544:   00000000 00000000 00000000 00000000
0x0000000111da9554:   00000000 00000000 00000000 00000000
0x0000000111da9564:   00000000 00000000 00000000 00000000
0x0000000111da9574:   00000000 00000000 00000000 00000000
0x0000000111da9584:   00000000 00000000 00000000 00000000
0x0000000111da9594:   00000000 00000000 00000000 00000000
0x0000000111da95a4:   00000000 00000000 00000000 00000000
0x0000000111da95b4:   00000000 00000000 00000000 00000000
0x0000000111da95c4:   00000000 00000000 00000000 00000000
0x0000000111da95d4:   00000000 00000000 00000000 00000000 


Stack slot to memory mapping:
stack at sp + 0 slots: 0x000000016f3f63e0 is pointing into the stack for thread: 0x000000015508d800
stack at sp + 1 slots: 0x0000000102a27758: _ZN12StubRoutines25_call_stub_return_addressE+0 in /Users/igyuseong/Library/Java/JavaVirtualMachines/corretto-17.0.7/Contents/Home/lib/server/libjvm.dylib at 0x0000000101dc0000
stack at sp + 2 slots: 0x000000016f3f5c90 is pointing into the stack for thread: 0x000000015508d800
stack at sp + 3 slots: {method} {0x00000008003f9650} 'setPriority' '(I)V' in 'java/lang/Thread'
stack at sp + 4 slots: 0x00000008003f9000 is pointing into metadata
stack at sp + 5 slots: {method} {0x00000008003f96a8} 'dumpStack' '()V' in 'java/lang/Thread'
stack at sp + 6 slots: 0x000000016f3f5b60 is pointing into the stack for thread: 0x000000015508d800
stack at sp + 7 slots: 0x0000000102580254: _ZN2os17is_readable_rangeEPKvS1_+0x2c in /Users/igyuseong/Library/Java/JavaVirtualMachines/corretto-17.0.7/Contents/Home/lib/server/libjvm.dylib at 0x0000000101dc0000

StubRoutines::SafeFetch32 [0x0000000111da94e0, 0x0000000111da94ec] (12 bytes)
[MachCode]
  0x0000000111da94e0: 0100 40b9 | e003 01aa | c003 5fd6 
[/MachCode]

---------------  P R O C E S S  ---------------

Threads class SMR info:
_java_thread_list=0x00006000000398e0, length=13, elements={
0x000000015508d800, 0x0000000155162000, 0x0000000155164800, 0x0000000155162c00,
0x0000000155168e00, 0x0000000155169400, 0x0000000155169a00, 0x000000015516a000,
0x0000000155172600, 0x0000000155166a00, 0x000000013308d800, 0x0000000155192c00,
0x00000001532be400
}

Java Threads: ( => current thread )
=>0x000000015508d800 JavaThread "main" [_thread_in_Java, id=8707, stack(0x000000016f1f4000,0x000000016f3f7000)]
  0x0000000155162000 JavaThread "Reference Handler" daemon [_thread_blocked, id=19459, stack(0x0000000170048000,0x000000017024b000)]
  0x0000000155164800 JavaThread "Finalizer" daemon [_thread_blocked, id=18947, stack(0x0000000170254000,0x0000000170457000)]
  0x0000000155162c00 JavaThread "Signal Dispatcher" daemon [_thread_blocked, id=31491, stack(0x0000000170578000,0x000000017077b000)]
  0x0000000155168e00 JavaThread "Service Thread" daemon [_thread_blocked, id=30979, stack(0x0000000170784000,0x0000000170987000)]
  0x0000000155169400 JavaThread "Monitor Deflation Thread" daemon [_thread_blocked, id=30723, stack(0x0000000170990000,0x0000000170b93000)]
  0x0000000155169a00 JavaThread "C2 CompilerThread0" daemon [_thread_in_native, id=23811, stack(0x0000000170b9c000,0x0000000170d9f000)]
  0x000000015516a000 JavaThread "C1 CompilerThread0" daemon [_thread_blocked, id=30211, stack(0x0000000170da8000,0x0000000170fab000)]
  0x0000000155172600 JavaThread "Sweeper thread" daemon [_thread_blocked, id=29955, stack(0x0000000170fb4000,0x00000001711b7000)]
  0x0000000155166a00 JavaThread "Common-Cleaner" daemon [_thread_blocked, id=24835, stack(0x00000001711c0000,0x00000001713c3000)]
  0x000000013308d800 JavaThread "C2 CompilerThread1" daemon [_thread_in_native, id=25347, stack(0x00000001713cc000,0x00000001715cf000)]
  0x0000000155192c00 JavaThread "JFR Recorder Thread" daemon [_thread_blocked, id=25603, stack(0x00000001715d8000,0x00000001717db000)]
  0x00000001532be400 JavaThread "JFR Periodic Tasks" daemon [_thread_blocked, id=29443, stack(0x00000001717e4000,0x00000001719e7000)]

Other Threads:
  0x0000000154150ee0 VMThread "VM Thread" [stack: 0x000000016fe3c000,0x000000017003f000] [id=18179]
  0x00000001541497a0 GCTaskThread "GC Thread#0" [stack: 0x000000016f400000,0x000000016f603000] [id=13059]
  0x0000000132f654b0 GCTaskThread "GC Thread#1" [stack: 0x00000001719f0000,0x0000000171bf3000] [id=29187]
  0x0000000132f65930 GCTaskThread "GC Thread#2" [stack: 0x0000000171bfc000,0x0000000171dff000] [id=28675]
  0x0000000132f65db0 GCTaskThread "GC Thread#3" [stack: 0x0000000171e08000,0x000000017200b000] [id=28419]
  0x0000000132f66620 GCTaskThread "GC Thread#4" [stack: 0x0000000172014000,0x0000000172217000] [id=27907]
  0x0000000132e3eb40 GCTaskThread "GC Thread#5" [stack: 0x0000000172220000,0x0000000172423000] [id=26883]
  0x0000000154149e50 ConcurrentGCThread "G1 Main Marker" [stack: 0x000000016f60c000,0x000000016f80f000] [id=13315]
  0x000000015414a6d0 ConcurrentGCThread "G1 Conc#0" [stack: 0x000000016f818000,0x000000016fa1b000] [id=14083]
  0x000000015414c540 ConcurrentGCThread "G1 Refine#0" [stack: 0x000000016fa24000,0x000000016fc27000] [id=16899]
  0x000000013385fdf0 ConcurrentGCThread "G1 Refine#1" [stack: 0x000000017242c000,0x000000017262f000] [id=27651]
  0x0000000152f04470 ConcurrentGCThread "G1 Refine#2" [stack: 0x0000000172638000,0x000000017283b000] [id=43011]
  0x0000000122e042c0 ConcurrentGCThread "G1 Refine#3" [stack: 0x0000000172844000,0x0000000172a47000] [id=33283]
  0x000000015414cde0 ConcurrentGCThread "G1 Service" [stack: 0x000000016fc30000,0x000000016fe33000] [id=21507]

Threads with active compile tasks:
C2 CompilerThread0      331 1058       4       jdk.jfr.internal.MetadataWriter::writeInt (148 bytes)
C2 CompilerThread1      331  854       4       jdk.internal.org.objectweb.asm.Frame::execute (2305 bytes)

VM state: not at safepoint (normal execution)

VM Mutex/Monitor currently owned by a thread: None

Heap address: 0x00000007e0000000, size: 512 MB, Compressed Oops mode: Zero based, Oop shift amount: 3

CDS archive(s) mapped at: [0x0000000800000000-0x0000000800bd4000-0x0000000800bd4000), size 12402688, SharedBaseAddress: 0x0000000800000000, ArchiveRelocationMode: 0.
Compressed class space mapped at: 0x0000000801000000-0x0000000841000000, reserved size: 1073741824
Narrow klass base: 0x0000000800000000, Narrow klass shift: 0, Narrow klass range: 0x100000000

GC Precious Log:
 CPUs: 8 total, 8 available
 Memory: 16384M
 Large Page Support: Disabled
 NUMA Support: Disabled
 Compressed Oops: Enabled (Zero based)
 Heap Region Size: 1M
 Heap Min Capacity: 8M
 Heap Initial Capacity: 256M
 Heap Max Capacity: 512M
 Pre-touch: Disabled
 Parallel Workers: 8
 Concurrent Workers: 2
 Concurrent Refinement Workers: 8
 Periodic GC: Disabled

...


---------------  S Y S T E M  ---------------

OS:
uname: Darwin 22.5.0 Darwin Kernel Version 22.5.0: Mon Apr 24 20:53:44 PDT 2023; root:xnu-8796.121.2~5/RELEASE_ARM64_T8103 arm64
OS uptime: 0 days 0:30 hours
rlimit (soft/hard): STACK 8176k/65520k , CORE 0k/infinity , NPROC 2666/4000 , NOFILE 10240/infinity , AS infinity/infinity , CPU infinity/infinity , DATA infinity/infinity , FSIZE infinity/infinity , MEMLOCK infinity/infinity , RSS infinity/infinity
load average: 2.24 2.17 2.88

CPU: total 8 (initial active 8) 0x61:0x0:0x1b588bb3:0, fp, simd, crc, lse

Memory: 16k page, physical 16777216k(533872k free), swap 0k(0k free)

vm_info: OpenJDK 64-Bit Server VM (17.0.7+7-LTS) for bsd-aarch64 JRE (17.0.7+7-LTS), built on Apr 14 2023 01:28:28 by "ec2user" with clang Apple LLVM 13.0.0 (clang-1300.0.29.30)

END.

```

i skip some contents because Comment is too long (maximum is 65536 characters)
if you want more contents i will send email

**Comment by navyxliu** [comment]
The problem starts from 'V  [libjvm.dylib+0x35ecd0]  AsyncGetCallTrace+0x1a4'
it tried to determine whether it is a valid method. It may not be a mapped region on macOS.

**Comment by navyxliu** [comment]
This bug has been in the JBS with good explanation: https://bugs.openjdk.org/browse/JDK-8307549

jdk has been fixed [JDK-8283326](https://bugs.openjdk.org/browse/JDK-8283326) and backported it to jdk17.

**Comment by navyxliu** [comment]
@aak2075 
Could you try it using a newer version of corretto-17 (17.0.9+)

**Comment by aak2075** [comment]
The issue has been resolved using corretto-17.0.9. Thank you :)
